hooks/hookexecution/executor.go

```diff
@@ -0,0 +1,134 @@
+package hookexecution
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/hooks"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+const (
+	EndpointAuction = "/openrtb2/auction"
+	EndpointAmp     = "/openrtb2/amp"
+)
+
+type StageExecutor interface {
+	ExecuteEntrypointStage(req *http.Request, body []byte) ([]byte, *RejectError)
+	ExecuteRawAuctionStage(body []byte) ([]byte, *RejectError)
+	ExecuteProcessedAuctionStage(req *openrtb2.BidRequest) *RejectError
+	ExecuteBidderRequestStage(req *openrtb2.BidRequest, bidder string) *RejectError
+	ExecuteRawBidderResponseStage(response *adapters.BidderResponse) *RejectError
+	ExecuteAllProcessedBidResponsesStage(responses []*adapters.BidderResponse) //TODO: check that responses is the necessary param
+	ExecuteAuctionResponseStage(response *openrtb2.BidResponse)
+}
+
+type HookStageExecutor interface {
+	StageExecutor
+	SetAccount(account *config.Account)
+	GetOutcomes() []StageOutcome
+}
+
+type HookExecutor struct {
+	InvocationCtx *hookstage.InvocationContext
+	Endpoint      string
+	PlanBuilder   hooks.ExecutionPlanBuilder
+	stageOutcomes []StageOutcome
+}
+
+func (executor *HookExecutor) SetAccount(account *config.Account) {
+	executor.InvocationCtx.Account = account
+	executor.InvocationCtx.AccountId = account.ID
```

Comment:
We set the `InvocationCtx` at the very beginning of the endpoint processing so it should never be `nil`, but we can add a guard condition just in case